src/FullCalendar.tsx

```diff
@@ -1,32 +1,22 @@
 import * as React from 'react'
 import { Calendar, OptionsInput } from '@fullcalendar/core'
 
-export default class FullCalendar extends React.Component<OptionsInput, any> {
+const { forwardRef, useEffect, useImperativeHandle, useRef, useState } = React
 
-  private elRef: any = React.createRef()
-  private calendar: Calendar
+export default forwardRef((props: OptionsInput, componentRef) => {
+  const elRef = useRef()
+  const [calendar, setCalendar] = useState()
```

Comment:
i think a ref might be more appropriate here.

https://reactjs.org/docs/hooks-reference.html#useref

> However, useRef() is useful for more than the ref attribute. It’s handy for keeping any mutable value around similar to how you’d use instance fields in classes.

Reply:
Sorry, I do not understand... 
1. `React.useRef` is the new `React.createRef`
2. `React.forwardRef` is necessary to expose `getApi` to parent component using ref

```
const calRef = useRef()
const onEventSelect = () => { const calApi = calRef.current.getApi(); /* Do something with api */ }
return <Calendar onEventSelect={onEventSelect} ref={calRef} />
```

Reply:
I'm not commenting about how you are doing forwardRef or doing anything with the `ref=""` attribute. I'm saying that instead of using `useState` you should be using `useRef`. `useRef` has another purpose than just supporting `ref=""`. It is another way to persist internal state. Read the article I posted near the the text:

> useRef() is useful for more than the ref attribute. It’s handy for keeping any mutable value around similar to how you’d use instance fields in classes.

If you reset the *state*, then it will trigger a second execution of the render function. If you use a ref for the internal state, it won't

Reply:
Alright. I'll check this out.